src/main/java/com/ea/gatling/AwsGatlingRunner.java

```diff
@@ -37,52 +37,72 @@ public AwsGatlingRunner() {
     }
 
     public Map<String, Instance> launchEC2Instances(String instanceType, int instanceCount, String ec2KeyPairName, String ec2SecurityGroup, String amiId) {
-        RunInstancesResult runInstancesResult = ec2client.runInstances(new RunInstancesRequest()
-                .withImageId(amiId)
-                .withInstanceType(instanceType)
-                .withMinCount(instanceCount)
-                .withMaxCount(instanceCount)
-                .withKeyName(ec2KeyPairName)
-                .withSecurityGroups(ec2SecurityGroup));
         Map<String, Instance> instances = new HashMap<String, Instance>();
 
-        for (Instance instance : runInstancesResult.getReservation().getInstances()) {
-            System.out.println(instance.getInstanceId() + " launched");
-            instances.put(instance.getInstanceId(), instance);
+        // Setup a filter to find any previously generated EC2 instances
+        Filter[] filters = new Filter[2];
+        filters[0] = new Filter("tag:Name").withValues(GATLING_NAME_TAG.getValue());
+        filters[1] = new Filter("instance-state-name").withValues("running");
+
+        DescribeInstancesResult describeInstancesResult = ec2client.describeInstances(new DescribeInstancesRequest().withFilters(filters));
+
+        // Check for existing EC2 instances that fit the filter criteria and use those
+        for (Reservation reservation : describeInstancesResult.getReservations()) {
+            for (Instance instance : reservation.getInstances()) {
+                System.out.println("Reservations " + instance.getInstanceId() + " (" + instance.getState().getName() + "): " + instance.getSecurityGroups().get(0).getGroupName());
```

Comment:
This gets pretty hard to read. Do you mind changing it to a System.out.format() call instead? See line 67 for an example.